src/Strategy/Conan/ConanGraph.hs

```diff
@@ -0,0 +1,310 @@
+    OtherContext Text
+  deriving (Show, Eq, Ord)
+
+instance FromJSON ConanGraphNodeContext where
+  parseJSON = withText "Context" $ \ctx -> do
+    case ctx of
+      "host" -> pure HostContext
+      "build" -> pure BuildContext
+      other -> pure $ OtherContext other
+
+-- | Conan's package type
+-- Doc: https://docs.conan.io/2.0/reference/conanfile/attributes.html#package-type
+data ConanPackageType
+  = -- The package is an application.
+    Application
+  | -- The package is a generic library.
+    Library
+  | -- The package is a shared library.
+    SharedLibrary
+  | -- The package is a static library.
+    StaticLibrary
+  | -- The package is a header only library.
+    HeaderLibrary
+  | -- The package only contains build scripts.
+    BuildScript
+  | -- The package is a python require.
+    PythonRequires
+  | -- The type of the package is unknown.
+    Unknown Text
+  deriving (Show, Eq, Ord)
+
+instance FromJSON ConanPackageType where
+  parseJSON = withText "ConanPackageType" $ \ctx -> do
+    case ctx of
+      "application" -> pure Application
+      "library" -> pure Library
+      "shared-library" -> pure SharedLibrary
+      "static-library" -> pure StaticLibrary
+      "header-library" -> pure HeaderLibrary
+      "build-scripts" -> pure BuildScript
+      "python-require" -> pure PythonRequires
+      other -> pure $ Unknown other
+
+indexById :: ConanGraph -> Map TextLike ConanGraphNode
+indexById = mempty
+
+mkGraph :: Has (Grapher Dependency) sig m => ConanGraph -> m ()
+mkGraph conanGraph = do
+  for_ (nodes conanGraph) $ \dep -> do
+    let resolvedDep = toDependency dep
+    if isDirectDep dep
+      then direct resolvedDep
+      else deep resolvedDep
+
+    let transitives = mapMaybe (refToDependency . TextLike) (keys $ requires dep)
+    for_ transitives $ \childDep -> do
+      deep childDep
+      edge resolvedDep childDep
+  where
+    registry :: Map TextLike ConanGraphNode
+    registry = indexById conanGraph
+
+    isDirectDep :: ConanGraphNode -> Bool
+    isDirectDep _ = True
+
+    refToDependency :: TextLike -> Maybe Dependency
+    refToDependency nodeId = toDependency <$> Data.Map.lookup nodeId registry
+
+toDependency :: ConanGraphNode -> Dependency
+toDependency cn =
+  Dependency
+    { dependencyType = ConanType
+    , dependencyName = name cn
+    , dependencyVersion = Just (CEq $ getVersion cn)
+    , dependencyLocations = getLocations cn
+    , dependencyEnvironments = Set.singleton $ getEnv cn
+    , dependencyTags = mempty
+    }
+
+getLocations :: ConanGraphNode -> [Text]
+getLocations cn = case packgeType cn of
+  SharedLibrary -> maybeToList $ buildFolder cn
+  _ -> maybeToList $ sourceFolder cn
+
+getEnv :: ConanGraphNode -> DepEnvironment
+getEnv ConanGraphNode{..} = case (test, context) of
+  (True, _) -> EnvTesting
+  (_, BuildContext) -> EnvDevelopment
+  (_, HostContext) -> EnvProduction
+  _ -> EnvProduction
+
+getVersion :: ConanGraphNode -> Text
```

Comment:
Without looking at the specs, I'm not sure what the output of this looks like or why we're doing it. Can you add a comment explaining what's going on and what the output from this function will look like?

Also, I think this is just our creation, but if this version is something that Conan specifies, let's link to that spec. If it is our creation, please note that in the comment